src/usermod.c

```diff
@@ -308,13 +308,13 @@ static struct ulong_range getulong_range(const char *str)
 	errno = 0;
 	first = strtoll(str, &pos, 10);
 	if (('\0' == *str) || ('-' != *pos ) || (ERANGE == errno) ||
-	    (first != (unsigned long int)first))
+	    (first != (long long)(unsigned long int)first))
```

Comment:
Would you please explain what the double cast is doing (and what the single cast was doing before it)?  This things aren't trivial, and I'd like to see an explanation of why this is correct, and what's the intention of the code.